demos/projects/ESPRESSIF/aziotkit/components/sample-azure-iot/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+set(ROOT_PATH
+    ${CMAKE_CURRENT_LIST_DIR}/../../../../../..
+)
+
+set(COMPONENT_SOURCES
+    ${ROOT_PATH}/demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c
+)
+
+idf_component_get_property(MBEDTLS_DIR mbedtls COMPONENT_DIR)
+
+list(APPEND COMPONENT_SOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/backoff_algorithm.c
+    ${CMAKE_CURRENT_LIST_DIR}/transport_tls_esp32.c
+    ${CMAKE_CURRENT_LIST_DIR}/crypto_esp32.c
+)
+
+set(COMPONENT_INCLUDE_DIRS
+    ${CMAKE_CURRENT_LIST_DIR}/../../config
+    ${CMAKE_CURRENT_LIST_DIR}
+    ${MBEDTLS_DIR}/mbedtls/include
+    ${ROOT_PATH}/demos/common/transport
+    ${ROOT_PATH}/demos/common/utilities
+    ${ROOT_PATH}/demos/sample_azure_iot_pnp
+)
+
+idf_component_register(
+    SRCS ${COMPONENT_SOURCES}
+    INCLUDE_DIRS ${COMPONENT_INCLUDE_DIRS}
+    REQUIRES mbedtls tcp_transport coreMQTT azure-sdk-for-c azure-iot-middleware-freertos)

```

Comment:
>coreMQTT azure-sdk-for-c azure-iot-middleware-freertos [](http://example.com/codeflow?start=35&length=54)

can this only depend upon middleware?

middleware depends upon coreMqtt and embedded sdk?